includes/base/widget-base.php

```diff
@@ -371,12 +372,20 @@ protected function get_initial_config() {
 			'upsale_data' => $this->get_upsale_data(),
 		];
 
+		$config['upsale_data'] = apply_filters( 'elementor/widgets/' . $this->get_name() . '/custom_promotion', $config['upsale_data'] ) ?? $this->get_upsale_data();
+		if ( isset( $config['upsale_data']['image'] ) ) {
+			sanitize_file_name( $config['upsale_data']['image'] );
+		}
+
 		$stack = Plugin::$instance->controls_manager->get_element_stack( $this );
 
 		if ( $stack ) {
 			$config['controls'] = $this->get_stack( false )['controls'];
 			$config['tabs_controls'] = $this->get_tabs_controls();
 		}
+		if ( isset( $config['upsale_data']['upgrade_url'] ) && false === Validate_Promotion::domain_is_on_elementor_dot_com( $config['upsale_data']['upgrade_url'] ) ) {
```

Comment:
why false ?

Reply:
The filter runs and this condition checks if the url returned from the filter is invalid and replaces the url with the default for this promotion in that case.